tests/unit/blocking_channel_tests.py

```diff
@@ -65,4 +66,25 @@ def test_basic_consume(self):
             self.obj.basic_consume(mock.Mock(), "queue")
 
             self.assertEqual(self.obj._consumer_infos['ctag0'].state,
-                             blocking_connection._ConsumerInfo.ACTIVE)
\ No newline at end of file
+                             blocking_connection._ConsumerInfo.ACTIVE)
+
+    def test_with(self):
```

Comment:
The combination of the word "with" with the other words in these new test names results in somewhat confusing names. For example, "test_with_close" raises the question "test what with close?" 😃 

It's desirable to use descriptive names, since that's what will show up in the test logs. So, the more information in the test name, the better. In tests, it's acceptable to use very long names, when needed. For example, `test_with_close`, might be more clear if named something like `test_context_manager_exit_with_closed_channel`. `test_with_raise` could be `test_context_manager_does_not_suppress_exception`, and `test_with` could be `test_context_manager`